frontend/src/metabase/visualizations/echarts/cartesian/model/axis.ts

```diff
@@ -461,13 +472,77 @@ export function getXAxisModel(
 
   const formatter = (value: RowValue) =>
     renderingContext.formatValue(value, {
-      column,
-      ...(settings.column?.(column) ?? {}),
+      column: dimensionModel.column,
+      ...(settings.column?.(dimensionModel.column) ?? {}),
       noRange: isHistogram,
     });
 
+  const xValues = dataset.map(datum => datum[X_AXIS_DATA_KEY]);
+
+  const timeSeriesInterval = getTimeSeriesXAxisInterval(
+    xValues,
+    rawSeries,
+    dimensionModel,
+  );
+  const numericInterval = getNumericXAxisInterval(
+    xValues,
+    dimensionModel,
+    settings,
+  );
   return {
     formatter,
     label,
+    timeSeriesInterval,
+    numericInterval,
   };
 }
+
+// We should always have results_timezone, but just in case we fallback to UTC
+export const DEFAULT_TIMEZONE = "Etc/UTC";
+
+function getTimezone(rawSeries: RawSeries) {
+  const { results_timezone } = rawSeries[0].data;
+  return results_timezone || DEFAULT_TIMEZONE;
+}
+
+export function getTimeSeriesXAxisInterval(
+  xValues: RowValue[],
+  rawSeries: RawSeries,
+  dimensionModel: DimensionModel,
+): TimeSeriesInterval {
+  // We need three pieces of information to define a timeseries range:
+  // 1. interval - it's really the "unit": month, day, etc
+  // 2. count - how many intervals per tick?
+  // 3. timezone - what timezone are values in? days vary in length by timezone
+  const unit = minTimeseriesUnit(
+    dimensionModel.columns.map(column => column.unit),
+  );
+  const timezone = getTimezone(rawSeries);
+  const { count, interval } = (computeTimeseriesDataInverval(xValues, unit) ?? {
+    count: 1,
+    interval: "day",
+  }) as Pick<TimeSeriesInterval, "count" | "interval">;
+  return { count, interval, timezone };
+}
+
+export function getNumericXAxisInterval(
```

Comment:
Let's remove this function from this PR, I don't see `numericInterval` being used anywhere.